compiler/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -127,6 +128,10 @@ impl<'a> Resolver<'a> {
         self.self_type = self_type;
     }
 
+    pub fn get_self_type(&mut self) -> Option<Type> {
+        self.self_type.clone()
+    }
```

Comment:
Let's avoid cloning here:
```suggestion
    pub fn get_self_type(&mut self) -> Option<&Type> {
        self.self_type.as_ref()
    }
```

Reply:
https://github.com/noir-lang/noir/pull/2652/commits/c64574d947358e8eacf300efb1d29370486aa3ea